test/integration/targets/ios_ntp/tests/cli/ntp_configuration.yaml

```diff
@@ -0,0 +1,42 @@
+---
+- name: Remove ntp (if set)
+  ios_ntp:
+    server: 10.75.32.5
+    source_int: Loopback0
+    acl: NTP_ACL
+    logging: true
+    key_id: 10
+    auth_key: 15435A030726242723273C21181319000A
+    auth: true
+    state: absent
+  ignore_errors: true
+
+- name: Reconfigure ntp as before
+  ios_ntp:
+    server: 10.75.32.5
+    source_int: Loopback0
+    acl: NTP_ACL
+    logging: true
+    key_id: 10
+    auth_key: 15435A030726242723273C21181319000A
+    auth: true
+    state: present
+
+- assert:
+    that:
+      - "result.changed == true"
+
+- name: Ensure idempotence
+  ios_ntp:
+    server: 10.75.32.5
+    source_int: Loopback0
+    acl: NTP_ACL
+    logging: true
+    key_id: 10
+    auth_key: 15435A030726242723273C21181319000A
+    auth: true
+    state: present
+
+- assert:
+    that:
+      - "result.changed == false"
```

Comment:
Your test isn't done yet. You should clean up the settings you have set on the device to ensure a clean run next time through.

Furthermore, look at https://github.com/ansible/ansible/blob/devel/test/integration/targets/nxos_ntp/tests/common/sanity.yaml It sets some config, sets other config, removes config, sets config again, and finally removes it at the end. And every step is repeated to check for idempotence. You aren't testing removal here because your removal statement at the top is expected to not need to run if you're on a clean slate (and nothing is ever being tested if `ignore_errors` is set). This needs to, at the very least, have a removal step at the end before it is complete

Note: In an attempt to not confuse further, `*true` and `*false` and so on aren't anything magic, this test just uses YAML anchors to avoid a lot of repetition. https://markusholtermann.eu/2016/04/reference-variables-in-ansible-yaml/ gives a decent overview of how they work

Reply:
**test_ios_ntp.py** 

Ok, I now understand that units test run against fixture file (where example config is stored). Three test run against fixture:
 
- Set configuration: in `set_module_args` we set different values compared to what we have in fixture. We also define a list of configuration commands that we expect to see in order to apply the new configuration. The list of expected commands is compared with a list of commands set by the module. If the 2 list match, the test pass.

-  Remove configuration: same as above but is valid for `no` commands

- Idempotent: Having the arguments in `set_module_args` the same as fixture, we expect no commands (empty list)

Reply:
**ntp_configuration.yaml** 

I have added more test using the pointers and deference - I was not aware about this functionality as well as about block. Useful indeed. I have called `*remove*` with `- always` to make sure the config is cleaned